src/Microsoft.OpenApi/Models/OpenApiDocument.cs

```diff
@@ -587,6 +587,21 @@ public static ReadResult Parse(string input,
         {
             return OpenApiModelFactory.Parse(input, format, settings);
         }
+        /// <summary>
+        /// Adds a schema to the components object of the current document.
+        /// </summary>
+        /// <param name="openApiSchema">The schema to add</param>
+        /// <param name="id">The id for the component</param>
+        /// <returns>Whether the schema was added to the components.</returns>
+        public bool AddComponentSchema(string id, OpenApiSchema openApiSchema)
```

Comment:
Is this not relevant for all types of components?  Should we just have a AddComponent method with overloads for the different types of components?

Reply:
I'm iterating here, so far the only one that was problematic in the context of yoko was the schemas.
We also need to think about making collections read only maybe, and adding a method to remove entries, so everything is in sync.